### PR TITLE
Implemented user account registration

### LINKED ISSUE #22: Feature request:  Account registering
**User Story**
For example: 
As a user
I want to able to create an account

**Priority**
[x]: High
[]: Medium
[]: Low

**Describe the solution you want to implement**
A way for a user to create an account and have the user's transactions being linked to this account.

**Describe alternatives you've considered**
Having no account system at all.

**Issue Dependencies**
#18 
#19 
**Additional context**
n/a

### PR DESCRIPTION
**Pull request by liappi**
Closes #22 

**Description**

User can now register an account with a username and password. The username and password is added to the `user` table in the DB.

**Testing**

Manual testing on Postman was done. A POST request was made with a username and password. On the first try the registration will succeed, and on subsequent tries the registration will fail because the account already exists in the database.

**Checklist**

- [X] I have tested my change
- [X] Code builds successfully and all tests pass
- [ ] Docs have been added/updated if needed

### CODE REVIEW

File: server/src/api/account.py [hunk 1]
```diff
@@ -24,13 +24,25 @@ def __init__(self, database):
 
     @cherrypy.expose
     def register(self):
-        """Called when client requests to register, should create the user on the user database, 
+        """Called when client requests to register, should create the user on the user table, 
             after creating a new user the client should be logged in
         
         Arguments:
             json -- check api doc for json format
```
Comment (lken274) [inline server/src/api/account.py]: nit: maybe add the actual json format to the docstring now that you have implemented the endpoint?

File: server/src/api/account.py [hunk 2]
```diff
@@ -24,13 +24,26 @@ def __init__(self, database):
...
-            json -- check api doc for json format
+            username (string): Name used for login
+            password (string): Password used for login
         """
-        return "2"
+        JSON_object = json.loads(cherrypy.request.body.read().decode('utf-8'))
+        username = JSON_object.get("username")
+        password = JSON_object.get("password")
+
+        try:
```
Comment (tant836) [inline server/src/api/account.py:38]: nit: Try catch not needed

**Review by tant836 — APPROVED** [review]
lgtm

**Review by lken274 — APPROVED** [review]
LGTM

**Pull request merged**